client/hijack.go

```diff
@@ -68,11 +68,11 @@ func (cli *Client) postHijacked(ctx context.Context, path string, query url.Valu
 	defer clientconn.Close()
 
 	// Server hijacks the connection, error 'connection closed' expected
-	clientconn.Do(req)
+	_, err = clientconn.Do(req)
```

Comment:
No, hijack is still necessary. The case is, server meets an error, it sends error message via Hijacked connection then close it. Client gets an ErrPersistError, but it can still read error message from low level tcp connection.
So we need to do Hijack() even though error happened, only need to throw it out and let client choose how to handle the error